### DISCUSSION TITLE
How to load tag id from rest api request?

### DISCUSSION THREAD

**Discussion by devbipu** [gtm-support/vue-gtm#407] [Q&A]
Hi, I'm using Vue.js with Laravel backend. I've option to set GTM id from backend database. Now how can I load the GTM id using a api request then set it to the createGtm()?

```
app.use(
  createGtm({
    id: 'GTM-xxxxkxxx',
    enabled: true,
    debug: true,
    vueRouter: router
  }),
);
```

**Comment by Shinigami92** [comment]
Not sure if I fully understand what you need

You can access the gtm.id by using e.g. the composition api `const gtm = useGtm(); gtm.id`
Or by exporting the id you passed to the `createGtm` function on your own.

However, setting the id is just exactly possible as you already demonstrated above.
Just get the id from your backend as you like and then pass it to the function. How you do it, is not part of the plugin :thinking:
